Kaa Storm integration/server/src/main/java/spout/FlumeSourceSpout.java

```diff
@@ -0,0 +1,229 @@
+
+
+package spout;
+
+import backtype.storm.spout.SpoutOutputCollector;
+import backtype.storm.task.TopologyContext;
+import backtype.storm.topology.IRichSpout;
+import backtype.storm.topology.OutputFieldsDeclarer;
+import backtype.storm.tuple.Values;
+import backtype.storm.utils.Utils;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.apache.flume.*;
+import org.apache.flume.instrumentation.SinkCounter;
+import org.apache.flume.node.MaterializedConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import common.MaterializedConfigurationProvider;
+import common.StormEmbeddedAgentConfiguration;
+import producer.AvroTupleProducer;
+import common.Constants;
+
+import java.util.List;
+import java.util.Map;
+
+
+@SuppressWarnings("serial")
+public class FlumeSourceSpout implements IRichSpout {
+	public static final String FLUME_PROPERTY_PREFIX = "flume-agent";
+	public static final String PROP_FLUME_AGENT_NAME = "flume-agent";
+	public static final String PROP_FLUME_BATCH_SIZE = "batch-size";
+	public static final int BATCH_SIZE = 100;
+
+	private Channel channel;
+	private SourceRunner sourceRunner;
+	private int batchSize = BATCH_SIZE;
+	private SinkCounter sinkCounter;
+	private SpoutOutputCollector outputCollector;
+	private static final Logger LOG = LoggerFactory.getLogger(FlumeSourceSpout.class);
+
+
+    public String getFlumePropertyPrefix() {
+		return FLUME_PROPERTY_PREFIX;
+	}
+
+	private AvroTupleProducer avroTupleProducer;
+
+	public AvroTupleProducer getAvroTupleProducer() {
+		return avroTupleProducer;
+	}
+
+	public void setAvroTupleProducer(AvroTupleProducer avroTupleProducer) {
+		this.avroTupleProducer = avroTupleProducer;
+	}
+
+	@SuppressWarnings("rawtypes")
+	public void open(Map config, TopologyContext context,
+			SpoutOutputCollector collector) {
+		MaterializedConfigurationProvider configurationProvider = new MaterializedConfigurationProvider();
+		
+		Map<String, String> flumeAgentProps = Maps.newHashMap();
+		for (Object key : config.keySet()) {
+			LOG.debug("Spout config:" + key.toString() + ":" + config.get(key));
+			if (key.toString().startsWith(getFlumePropertyPrefix())) {
+				if (key.toString().contains(PROP_FLUME_BATCH_SIZE)) {
+					String batchSizeStr = (String) config.get(key);
+					try {
+						this.batchSize = Integer.parseInt(batchSizeStr);
+					} catch (Exception e) {
+						this.batchSize = BATCH_SIZE;
+					}
+				} else {
+					flumeAgentProps.put(key.toString().replace(getFlumePropertyPrefix() + ".",""),
+																(String) config.get(key));
+				}
+			}
+		}
+
+		flumeAgentProps = StormEmbeddedAgentConfiguration.configure(
```

Comment:
Maybe make a static method instead of separate class that is used only in one place?